### PR TITLE
Add support for sentry logging

### PR DESCRIPTION
**Pull request by sapkra**
This is adding the possibility to track warnings and errors for jicofo, jigasi and jvb with [Sentry](https://sentry.io).

depends on:
* https://github.com/jitsi/jigasi/pull/247
* https://github.com/jitsi/jicofo/pull/449
* https://github.com/jitsi/jitsi-videobridge/pull/1152

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -459,6 +464,18 @@ succeed. If your users are coming in over the Internet (and not over LAN), this
...
+
+To activate and configure Sentry you have to set the `SENTRY_DSN` environment variable for the containers you want to monitor. Each container should have it's own DSN. More information including all configartion options you can find [here](https://docs.sentry.io/clients/java/config).
+
+You can also set a release and environment to filter for possible errors and warnings more easily. These can be configured by defining `SENTRY_ENVIRONMENT` and `SENTRY_RELEASE` environment variables.
+
+Currently available for:
+* JVB
+* Jicofo
+* Jigasi
+
```
Comment (luixxiul) [inline README.md:478]: These change need to be applied to https://github.com/jitsi/handbook/blob/master/docs/devops-guide/docker.md (cf: #601)

Comment (sapkra) [reply README.md]: https://github.com/jitsi/handbook/pull/18

**Comment by saghul** [comment]
All the component PRs landed. I'll land this for the next stable release.

**Comment by sapkra** [comment]
@saghul Thank your for reviewing and merging my PRs.

**Comment by saghul** [comment]
@sapkra Hey Paul, a new stable is out, can you give it a try and let me know if we are good to merge this one now?

**Comment by sapkra** [comment]
@saghul It's currently a very stressful time. I will find some time to test it next month.

**Comment by saghul** [comment]
No worries, take you time!

**Comment by saghul** [comment]
@sapkra Approved it. This is now ready to merge, right?

**Comment by sapkra** [comment]
I've still not tested it. I've just rebased all my branches.

**Comment by sapkra** [comment]
@saghul After fixing the variable names 2 days ago everything seems to work fine in my test environment.

**Comment by sapkra** [comment]
@saghul Any news about this one? Using it on production for month and everything works fine.

**Comment by saghul** [comment]
Oops I forgot about it. Will look at it this week. Last I checked nothing stood out!

**Comment by sapkra** [comment]
@saghul Fixed the conflict. Ready to be merged.

File: jicofo/rootfs/etc/cont-init.d/10-config [hunk 2]
```diff
@@ -14,7 +16,7 @@ fi
 tpl /defaults/jicofo.conf > /config/jicofo.conf
 
 if [[ ! -f /config/logging.properties ]]; then
-    cp /defaults/logging.properties /config
+    tpl /defaults/logging.properties > /config/logging.properties
```
Comment (saghul) [inline jicofo/rootfs/etc/cont-init.d/10-config:19]: Let's template the file unconditionally?

File: jigasi/rootfs/etc/cont-init.d/10-config [hunk 3]
```diff
@@ -17,7 +19,7 @@ if [[ -f /config/custom-sip-communicator.properties ]]; then
 fi
 
 if [[ ! -f /config/logging.properties ]]; then
-    cp /defaults/logging.properties /config
+    tpl /defaults/logging.properties > /config/logging.properties
```
Comment (saghul) [inline jigasi/rootfs/etc/cont-init.d/10-config:22]: ditto

File: jvb/rootfs/etc/cont-init.d/10-config [hunk 4]
```diff
@@ -21,7 +22,7 @@ fi
 tpl /defaults/jvb.conf > /config/jvb.conf
 
 if [[ ! -f /config/logging.properties ]]; then
-    cp /defaults/logging.properties /config
+    tpl /defaults/logging.properties > /config/logging.properties
```
Comment (saghul) [inline jvb/rootfs/etc/cont-init.d/10-config:25]: ditto

**Comment by saghul** [comment]
3 minor comments, LGTM otherwise!

File: jicofo/rootfs/etc/cont-init.d/10-config (hunk 2 above)
Comment (sapkra) [reply jicofo/rootfs/etc/cont-init.d/10-config]: I added a commit to remove the conditions.

**Review by saghul — APPROVED** [review]

**Comment by saghul** [comment]
Heads up, I'm reverting this. Looks like it ends up being enabled by default:

```
WARNING: The JICOFO_SENTRY_DSN variable is not set. Defaulting to a blank string.
WARNING: The JVB_SENTRY_DSN variable is not set. Defaulting to a blank string.

...
jicofo_1   | Apr 17, 2021 8:40:06 PM org.jitsi.utils.logging2.LoggerImpl log
jicofo_1   | WARNING: Disabling TLS certificate verification!
jicofo_1   | java.util.logging.ErrorManager: 1: An exception occurred while creating a new event in Sentry
jicofo_1   | java.lang.IllegalArgumentException: Illegal character in path at index 0: ""
jicofo_1   | 	at java.net.URI.create(URI.java:852)
jicofo_1   | 	at io.sentry.dsn.Dsn.<init>(Dsn.java:41)
jicofo_1   | 	at io.sentry.SentryClientFactory.instantiateFrom(SentryClientFactory.java:114)
jicofo_1   | 	at io.sentry.SentryOptions.<init>(SentryOptions.java:44)
jicofo_1   | 	at io.sentry.SentryOptions.from(SentryOptions.java:89)
jicofo_1   | 	at io.sentry.SentryOptions.defaults(SentryOptions.java:116)
jicofo_1   | 	at io.sentry.SentryOptions.defaults(SentryOptions.java:100)
jicofo_1   | 	at io.sentry.Sentry.getStoredClient(Sentry.java:168)
jicofo_1   | 	at io.sentry.Sentry.capture(Sentry.java:256)
jicofo_1   | 	at io.sentry.jul.SentryHandler.publish(SentryHandler.java:105)
jicofo_1   | 	at java.util.logging.Logger.log(Logger.java:738)
jicofo_1   | 	at org.jitsi.utils.logging2.LoggerImpl.log(LoggerImpl.java:129)
jicofo_1   | 	at org.jitsi.utils.logging2.LoggerImpl.warn(LoggerImpl.java:267)
jicofo_1   | 	at org.jitsi.impl.protocol.xmpp.XmppProviderImpl.createXmppConnection(XmppProviderImpl.java:168)
jicofo_1   | 	at org.jitsi.impl.protocol.xmpp.XmppProviderImpl.<init>(XmppProviderImpl.java:109)
jicofo_1   | 	at org.jitsi.jicofo.JicofoServices$createXmppProviderFactory$1.createXmppProvider(JicofoServices.kt:85)
jicofo_1   | 	at org.jitsi.jicofo.xmpp.XmppServices.<init>(XmppServices.kt:34)
jicofo_1   | 	at org.jitsi.jicofo.JicofoServices.<init>(JicofoServices.kt:108)
jicofo_1   | 	at org.jitsi.jicofo.Main.main(Main.java:68)
```

(and the same for the JVB).

@sapkra ccan you please take a look when you get a chance?

**Comment by sapkra** [comment]
@saghul It's seems like a warning of docker-compose because I'm mapping the env vars `SENTRY_DSN="${JICOFO_SENTRY_DSN}"`. I have no idea how I could prevent it, except from mapping them inside the container, but then it's not that clean anymore.

**Comment by saghul** [comment]
Maybe we can give it a default value in the compose file? And then compare it in the containers...

**Comment by sapkra** [comment]
@saghul Fixed it in #1024

**Pull request merged**